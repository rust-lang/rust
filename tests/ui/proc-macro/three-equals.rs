//@ proc-macro: three-equals.rs

extern crate three_equals;

use three_equals::three_equals;

fn main() {
    // This one is okay.
    three_equals!(===);

    // Need exactly three equals.
    three_equals!(==); //~ ERROR found 2 equal signs, need exactly 3

    // Need exactly three equals.
    three_equals!(=====); //~ ERROR expected EOF

    // Only equals accepted.
    three_equals!(abc); //~ ERROR expected `=`

    // Only equals accepted.
    three_equals!(!!); //~ ERROR expected `=`

    // Only three characters expected.
    three_equals!(===a); //~ ERROR expected EOF
}
