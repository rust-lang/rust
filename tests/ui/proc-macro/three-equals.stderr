error: found 2 equal signs, need exactly 3
  --> $DIR/three-equals.rs:12:5
   |
LL |     three_equals!(==);
   |     ^^^^^^^^^^^^^^^^^
   |
   = help: input must be: `===`
   = note: this error originates in the macro `three_equals` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected EOF, found `=`.
  --> $DIR/three-equals.rs:15:22
   |
LL |     three_equals!(=====);
   |                      ^
   |
note: last good input was here
  --> $DIR/three-equals.rs:15:21
   |
LL |     three_equals!(=====);
   |                     ^
   = help: input must be: `===`

error: expected `=`, found `abc`.
  --> $DIR/three-equals.rs:18:19
   |
LL |     three_equals!(abc);
   |                   ^^^

error: expected `=`, found `!`.
  --> $DIR/three-equals.rs:21:19
   |
LL |     three_equals!(!!);
   |                   ^

error: expected EOF, found `a`.
  --> $DIR/three-equals.rs:24:22
   |
LL |     three_equals!(===a);
   |                      ^
   |
note: last good input was here
  --> $DIR/three-equals.rs:24:21
   |
LL |     three_equals!(===a);
   |                     ^
   = help: input must be: `===`

error: aborting due to 5 previous errors

